tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -1652,71 +1781,76 @@ def test_internal_project(self):
         ).exists()
 
 
-@pytest.mark.parametrize(
-    "condition",
-    (
-        {"op": "and", "inner": []},
-        {"op": "and", "inner": [{"op": "and", "inner": []}]},
-        {"op": "or", "inner": []},
-        {"op": "or", "inner": [{"op": "or", "inner": []}]},
-        {"op": "not", "inner": {"op": "or", "inner": []}},
-        {"op": "eq", "ignoreCase": True, "name": "field1", "value": ["val"]},
-        {"op": "eq", "name": "field1", "value": ["val"]},
-        {"op": "glob", "name": "field1", "value": ["val"]},
-    ),
-)
-def test_condition_serializer_ok(condition):
-    serializer = DynamicSamplingConditionSerializer(data=condition)
-    assert serializer.is_valid()
-    assert serializer.validated_data == condition
-
-
-@pytest.mark.parametrize(
-    "condition",
-    (
-        {"inner": []},
-        {"op": "and"},
-        {"op": "or"},
-        {"op": "eq", "value": ["val"]},
-        {"op": "eq", "name": "field1"},
-        {"op": "glob", "value": ["val"]},
-        {"op": "glob", "name": "field1"},
-    ),
-)
-def test_bad_condition_serialization(condition):
-    serializer = DynamicSamplingConditionSerializer(data=condition)
-    assert not serializer.is_valid()
-
-
-def test_rule_config_serializer():
-    data = {
-        "rules": [
-            {
-                "sampleRate": 0.7,
-                "type": "trace",
-                "active": False,
-                "id": 1,
-                "condition": {
-                    "op": "and",
-                    "inner": [
-                        {"op": "eq", "name": "field1", "value": ["val"]},
-                        {"op": "glob", "name": "field1", "value": ["val"]},
-                    ],
+class TestDynamicSamplingSerializers(BaseTestCase):
+    @pytest.mark.parametrize(
+        "condition",
+        (
+            {"op": "and", "inner": []},
+            {"op": "and", "inner": [{"op": "and", "inner": []}]},
+            {"op": "or", "inner": []},
+            {"op": "or", "inner": [{"op": "or", "inner": []}]},
+            {"op": "not", "inner": {"op": "or", "inner": []}},
+            {"op": "eq", "ignoreCase": True, "name": "field1", "value": ["val"]},
+            {"op": "eq", "name": "field1", "value": ["val"]},
+            {"op": "glob", "name": "field1", "value": ["val"]},
+        ),
+    )
+    def test_condition_serializer_ok(self, condition):
+        serializer = DynamicSamplingConditionSerializer(data=condition)
+        assert serializer.is_valid()
+        assert serializer.validated_data == condition
+
+    @pytest.mark.parametrize(
```

Comment:
It does actually work, at least from my testing:
```shell
test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition0] PASSED [ 12%]{'op': 'and', 'inner': []}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition1] PASSED [ 25%]{'op': 'and', 'inner': [{'op': 'and', 'inner': []}]}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition2] PASSED [ 37%]{'op': 'or', 'inner': []}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition3] PASSED [ 50%]{'op': 'or', 'inner': [{'op': 'or', 'inner': []}]}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition4] PASSED [ 62%]{'op': 'not', 'inner': {'op': 'or', 'inner': []}}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition5] PASSED [ 75%]{'op': 'eq', 'ignoreCase': True, 'name': 'field1', 'value': ['val']}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition6] PASSED [ 87%]{'op': 'eq', 'name': 'field1', 'value': ['val']}

test_project_details.py::TestDynamicSamplingSerializers::test_condition_serializer_ok[condition7] PASSED [100%]{'op': 'glob', 'name': 'field1', 'value': ['val']}
```

Reply:
But this is because `BaseTestCase` doesn't have anything to do with `unittest.TestCase`. The inheritance tree is something like:
.
└── BaseTestCase/
    ├── Fixtures
    └── Exam/
        └── AssertsMixin

Reply:
Well that explains how it is all working :smile: